### PR TITLE
Introduce HashTable containing 'bcType tag'-> 'node list' association.

### PR DESCRIPTION
**Pull request by dsteinmo**
This way we know which nodes are 'Wall', 'Inflow', etc.

### CODE REVIEW

File: include/Types.hpp [hunk 1]
```diff
@@ -27,4 +28,5 @@ namespace blitzdg {
     using index_tensor3_type = tensor3_type<index_type>;
     using real_tensor3_type = tensor3_type<real_type>;
 
+    using index_hashmap = std::unordered_map<blitzdg::index_type, blitzdg::index_vector_type>;
 }
```
Comment (k7miller) [inline include/Types.hpp:33]: Is this a type you intend on using elsewhere? It's not really a fundamental scalar, vector, or matrix type, so I am not sure that it belongs here.

Comment (dsteinmo) [reply include/Types.hpp:33]: I went back and forth on this one. It's going to be needed in the solvers themselves to pull in the boundary conditions hashmap, so I thought putting it in the blitzdg namespace made sense. Will think about it some more though...

File: src/test/TriangleNodesProvisionerTests.cpp [hunk 2]
```diff
@@ -400,26 +400,32 @@ namespace blitzdg {
...
 
-				index_vector_type resM(480), resP(480), resB(64), resmB(64);
+
+				index_vector_type resM(480), resP(480), resB(64), resmB(64), resmapW(64);
 				
 				resM =  vmapM - expectedvmapM;
 				resP =  vmapP - expectedvmapP;
 				resB =  vmapB - expectedvmapB;
 				resmB =  mapB - expectedmapB;
+				resmapW = bcMap.at(3) - expectedmapW;
```
Comment (dsteinmo) [inline src/test/TriangleNodesProvisionerTests.cpp]: TODO, enum for Wall =3, ... etc.

File: include/Types.hpp (hunk 1 above)
Comment (k7miller) [inline include/Types.hpp:33]: You don't need to include the blitzdg:: namespaces as you are already in that namespace.

File: src/TriangleNodesProvisioner.cpp [hunk 3]
```diff
@@ -560,6 +563,40 @@ namespace blitzdg {
...
+        index_matrix_type boundaryNodesMat(NumFacePoints, NumFaces*NumElements);
+
+        index_vector_type ones(NumFacePoints);
+        ones = 0*ii + 1;
+        boundaryNodesMat = ones(ii)*bcVec(jj);
+
+        index_type count=0;
+        for (index_type f=0; f < NumFaces*NumElements; ++f) {
+            for (index_type n=0; n < NumFacePoints; ++n) {
+                index_type bct = boundaryNodesMat(n, f);
```
Comment (k7miller) [inline src/TriangleNodesProvisioner.cpp:588]: Since `boundaryNodesMat` is row-oriented you should have the n-loop as the outer loop and the f-loop as the inner loop. Alternatively, a better approach would be to use an iterator to iterate over the elements of `boundaryNodesMat` in a row-contiguous manner.

File: src/TriangleNodesProvisioner.cpp [hunk 4]
```diff
@@ -12,6 +12,7 @@
 #include <cmath>
 #include <limits>
 #include <memory>
+#include <unordered_map>
```
Comment (k7miller) [inline src/TriangleNodesProvisioner.cpp:15]: You don't need this include.

File: src/TriangleNodesProvisioner.cpp (hunk 3 above)
Comment (k7miller) [reply src/TriangleNodesProvisioner.cpp:588]: Something like the following should work:
```
for (auto itr = boundaryNodesMat.begin(); itr != boundaryNodesMat.end(); ++itr)
```
